web/src/components/Terminal.tsx

```diff
@@ -0,0 +1,480 @@
+Compressing objects: 100% (4/4), done.
+Writing objects: 100% (4/4), 481 bytes | 481.00 KiB/s, done.
+Total 4 (delta 2), reused 0 (delta 0), pack-reused 0
+remote: Resolving deltas: 100% (2/2), completed with 1 local object.
+To https://github.com/gibsonbailey/git-streak-tracker.git
+   3e6f0d6..89abcdef  main -> main`
+
+const Content = () => {
+    const [ mode, setMode ] = useState<'shell' | 'vim'>('shell')
+    const [ linesChanged, setLinesChanged ] = useState(0)
+    const [ currentCommand, setCurrentCommand ] = useState('')
+    const [ commandOutputs, setCommandOutputs ] = useState<string[]>([])
+    const [ currentCommandCursorIndex, setCurrentCommandIndex ] = useState(0)
+    const [ commandSpeed, setCommandSpeed ] = useState<'low' | 'high'>('low')
+
+
+    useEffect(() => {
+        setTimeout(() => {
+            setCurrentCommand('ls -la')
+        }, 1500)
+    }, [])
+
+    const vimModeFinished = () => {
+        setMode('shell')
+        setTimeout(() => {
+            setCommandSpeed('high')
+            setCurrentCommand('git commit -m "Minor change to language."')
+            setCurrentCommandIndex(0)
+        }, 1000)
+    }
+
+    useEffect(() => {
+        if (currentCommand == '') {
+            return
+        }
+        if (currentCommandCursorIndex < currentCommand.length) {
+            const timeout = setTimeout(() => {
+                setCurrentCommandIndex(currentCommandCursorIndex + 1)
+            }, Math.floor((commandSpeed === 'low' ? 50 : 20) + (Math.random() * 30)))
+            return () => { clearTimeout(timeout) }
+        } else {
+            if (currentCommand === 'ls -la') {
+                setTimeout(() => {
+                    setCurrentCommand('')
+                    setCommandOutputs([ lsOutput ])
+                }, 200)
+                setTimeout(() => {
+                    setCurrentCommand('vim code.tsx')
+                    setCurrentCommandIndex(0)
+                }, 1200)
+            } else if (currentCommand === 'vim code.tsx') {
+                setTimeout(() => {
+                    setMode('vim')
+                    setCurrentCommand('')
+                    setLinesChanged(1)
+                }, 200)
+            } else if (currentCommand === 'git commit -m "Minor change to language."') {
+                setTimeout(() => {
+                    setCommandSpeed('low')
+                    setLinesChanged(0)
+                    setCurrentCommand('')
+                    setCommandOutputs([ lsOutput, gitCommitOutput ])
+                }, 200)
+                setTimeout(() => {
+                    setCurrentCommand('git push')
+                    setCurrentCommandIndex(0)
+                }, 1000)
+            } else if (currentCommand === 'git push') {
+                setTimeout(() => {
+                    setCurrentCommand('')
+                    setCommandOutputs([ lsOutput, gitCommitOutput, gitPushOutput ])
+                }, 200)
```

Comment:
Lets add some newlines in here to make it more readable

Reply:
✅